service/tbc/peer/rawpeer/rawpeer.go

```diff
@@ -110,8 +110,8 @@ func (r *RawPeer) Id() int {
 
 func (r *RawPeer) Write(timeout time.Duration, msg wire.Message) error {
 	r.mtx.Lock()
+	defer r.mtx.Unlock()
 	conn := r.conn
-	r.mtx.Unlock()
```

Comment:
There is no way this is right compared to the defer. With the defer we execute everything in the same shot. With this, we conn can and will be intertwined. We need more diagnostics on this.

Reply:
Derp, I read that exactly the opposite way.